### ISSUE TITLE
test examples

### ISSUE THREAD

**Issue by alekspickle** [seanmonstar/warp#637]
I wanted to try warp in my next interview job task, and I love first working experience with warp, really. Unfortunately I ended up rewriting it all to actix, just because there is not enough examples featuring tests for a warp server, such as post a specific body and expect such and such output. I was confused how to do that.

Maybe there is such examples and I am just not patient enough to find them, in this case I will much appreciate the link!
But with those examples warp could gain more community weight, and therefore more people willing to help its development! 
I'll definitely invest more time to analyze how it is possible, but if someone can curb out of real apps some MVRE it would be great!

**Comment by jxs** [comment]
Hi, did you look into the [test docs](https://docs.rs/warp/0.2.3/warp/test/index.html)? what examples do you find lacking?

**Comment by alekspickle** [comment]
Yes I looked at these. My issue goes from not understanding where exactly am I launching the very request and how is it tied to my server specifically.

**Comment by jxs** [comment]
you need to create a test and use `warp::test::request` to test your filter.
As the doc shows (in this case it's a filter called sum):
```Rust
use warp::Filter;

fn sum() -> impl Filter<Extract = (u32,), Error = warp::Rejection> + Copy {
    warp::path::param()
        .and(warp::path::param())
        .map(|x: u32, y: u32| {
            x + y
        })
}

#[test]
fn test_sum() {
    let filter = sum();

    // Execute `sum` and get the `Extract` back.
    let value = warp::test::request()
        .path("/1/2")
        .filter(&filter)
        .unwrap();
    assert_eq!(value, 3);

    // Or simply test if a request matches (doesn't reject).
    assert!(
        !warp::test::request()
            .path("/1/-5")
            .matches(&filter)
    );
}
```

**Comment by calvinbrown085** [comment]
@jxs I recently ran into a problem where I needed `#[tokio::test]` instead, should those be added?

**Comment by alekspickle** [comment]
test docs are cool, but having examples is way easier to find

Anyway, since I opened this issue, you added a lot of examples, thank you!
